File: examples/governance_static_go/domain/activate_policy.go
package domain

import (
	"time"
	"fmt"
)

type ActivatePolicy struct {
	PolicyId string `json:"policy_id"`
	EffectiveDate time.Time `json:"effective_date"`
}

func (c ActivatePolicy) CommandName() string { return "ActivatePolicy" }

func (c ActivatePolicy) Execute(repo GovernancePolicyRepository) (*GovernancePolicy, *ActivatedPolicy, error) {
	existing, err := repo.Find(c.PolicyId)
	if err != nil {
		return nil, nil, err
	}
	if existing == nil {
		return nil, nil, fmt.Errorf("GovernancePolicy not found: %s", c.PolicyId)
	}
	existing.EffectiveDate = c.EffectiveDate
	if existing.Status != "draft" {
		return nil, nil, fmt.Errorf("cannot ActivatePolicy: GovernancePolicy is in %s state", existing.Status)
	}
	existing.Status = "active"
	existing.UpdatedAt = time.Now()
	if err := existing.Validate(); err != nil {
		return nil, nil, err
	}
	if err := repo.Save(existing); err != nil {
		return nil, nil, err
	}
	event := ActivatedPolicy{
		AggregateID: existing.ID,
		PolicyId: c.PolicyId,
		EffectiveDate: c.EffectiveDate,
		OccurredAt: time.Now(),
	}
	return existing, &event, nil
}

File: examples/governance/compliance_domain/lib/compliance_domain/governance_policy/commands/activate_policy.rb
module ComplianceDomain
  class GovernancePolicy
    module Commands
      class ActivatePolicy
        include Hecks::Command
        emits "ActivatedPolicy"

        attr_reader :policy_id, :effective_date

        def initialize(policy_id: nil, effective_date: nil)
          @policy_id = policy_id
          @effective_date = effective_date
        end

        def call
          existing = repository.find(policy_id)
          if existing
            unless existing.status == "draft"
              raise ComplianceDomain::Error, "Cannot ActivatePolicy: status must be 'draft', got '#{existing.status}'"
            end
            GovernancePolicy.new(
              id: existing.id,
              name: existing.name,
              description: existing.description,
              category: existing.category,
              framework_id: existing.framework_id,
              effective_date: effective_date,
              review_date: existing.review_date,
              requirements: existing.requirements,
              status: "active"
            )
          else
            raise ComplianceDomain::Error, "GovernancePolicy not found: #{policy_id}"
          end
        end
      end
    end
  end
end
